Use mlocked KES internally (#1374)
IntersectMBO/ouroboros-consensus: ouroboros-consensus-diffusion/changelog.d/20250130_100651_tdammers_mlocked_kes_rebase.md

### Non-Breaking

> - Use registry to ensure block forging threads are finalized in node kernel.

Reviewer: "Use registry" is a bit ambiguous. I would instead say
```suggestion
- Ensure that block forging threads erase their keys on shutdown in node kernel.
```

Author: It's not necessarily just erasing; finalizing may also involve shutting down a KES agent connection in the future.

### Non-Breaking

- Use registry to ensure block forging threads are finalized in node kernel.
- Tests have been adapted to match the API changes in

>   `ouroboros-consensus` and `ouroboros-consensus-cardano`.

Author: Fair enough.
